viabel/optimization.py

```diff
@@ -82,6 +82,8 @@ def optimize(self, n_iters, objective, init_param):
                     object_val, object_grad = objective(variational_param)
                     descent_dir = self.descent_direction(object_grad)
                     variational_param -= self._learning_rate * descent_dir
+                    if len(variational_param.shape) == 2:
```

Comment:
This looks very heuristic. What's the reason for adding it? Also, instead of `len(variational_param.shape)` you can use `variational_param.ndim`

Reply:
When training neural networks, people usually use weight decay (L2 regularization). I'm not sure if that is the case in variational inference, but I ran into numerical instability when fitting NVPFlow. That's the reason to add it. 

If it's fair to add weight decay to optimizers, I can add it as a parameter for all optimizers.

Reply:
Yeah, why don't you add a weight decay option to the optimizer (as an argument to `__init__`).

Reply:
Added.